cmd/oras/internal/display/metadata/tree/discover.go

```diff
@@ -66,18 +73,118 @@ func (h *discoverHandler) OnDiscovered(referrer, subject ocispec.Descriptor) err
 	}
 	referrerNode := node.AddPath(referrer.ArtifactType, referrer.Digest)
 	if h.verbose {
-		for k, v := range referrer.Annotations {
+		addDescriptorInfo(referrer, referrerNode)
+
+		if err := addAnnotationsInfo(referrer, referrerNode); err != nil {
+			return err
+		}
+
+		if err := addLayersInfo(referrer, referrerNode, ctx, target, platform); err != nil {
+			return err
+		}
+	}
+	h.nodes[referrer.Digest] = referrerNode
+	return nil
+}
+
+func addDescriptorInfo(referrer ocispec.Descriptor, referrerNode *tree.Node) {
+	descriptorNode := referrerNode.AddPath("Descriptor")
+	descriptorNode.AddPath(strings.TrimSpace("MediaType: " + referrer.MediaType))
+	descriptorNode.AddPath(strings.TrimSpace("Size: " + strconv.FormatInt(referrer.Size, 10)))
+}
+
+func addAnnotationsInfo(referrer ocispec.Descriptor, referrerNode *tree.Node) error {
+	if len(referrer.Annotations) != 0 {
+		annotationsNode := referrerNode.AddPath("Annotations")
+		annotationsKeys := sortedKeys(referrer.Annotations)
+		for _, k := range annotationsKeys {
+			v := referrer.Annotations[k]
 			bytes, err := yaml.Marshal(map[string]string{k: v})
 			if err != nil {
 				return err
 			}
-			referrerNode.AddPath(strings.TrimSpace(string(bytes)))
+			annotationsNode.AddPath(strings.TrimSpace(string(bytes)))
 		}
 	}
-	h.nodes[referrer.Digest] = referrerNode
 	return nil
 }
 
+func addLayersInfo(referrer ocispec.Descriptor, referrerNode *tree.Node, ctx context.Context, target oras.ReadOnlyTarget, platform option.Platform) error {
+	layersNode := tree.New("Layers")
+	layersDumped, ok := dumpLayers(referrer.Digest, layersNode, ctx, target, platform)
+	if ok != nil {
+		return ok
+	} else if layersDumped {
+		referrerNode.AddNode(layersNode)
+	}
+	return nil
+}
+
+func dumpLayers(digest digest.Digest, node *tree.Node, ctx context.Context, target oras.ReadOnlyTarget, platform option.Platform) (bool, error) {
+	fetchOpts := oras.DefaultFetchBytesOptions
+	fetchOpts.TargetPlatform = platform.Platform
+	_, content, err := oras.FetchBytes(ctx, target, digest.String(), fetchOpts)
+	if err != nil {
+		return false, fmt.Errorf("failed to fetch the content of %q: %w", "", err)
+	}
+
+	var jsonMap map[string]any
+	if err = json.Unmarshal(content, &jsonMap); err != nil {
+		return false, err
+	}
+
+	if layers, ok := jsonMap["layers"].([]interface{}); ok {
+		if len(layers) != 0 {
```

Comment:
This whole thing seems to me like a method fetchLayers or something which returns layers or error